tango/integrations/torch/train.py

```diff
@@ -180,24 +189,19 @@ def run(  # type: ignore[override]
 
         """
         # Validate device(s).
-        if torch.cuda.is_available():
-            if devices is None:
-                self.logger.info("CUDA is available")
-            elif all((x >= 0 for x in devices)):
-                if torch.cuda.device_count() < len(set(devices)):
-                    raise ConfigurationError(
-                        f"Only found {torch.cuda.device_count()} CUDA devices, "
-                        f"but you specified {len(set(devices))} device IDs"
-                    )
-            elif not all((x == -1 for x in devices)):
-                raise ConfigurationError("Invalid value for 'devices'")
+        if device_count <= 0:
+            raise ConfigurationError("Invalid value for 'device_count'. Must be at least 1.")
```

Comment:
```suggestion
            raise ConfigurationError("Invalid value for 'device_count'. Must be at least 0.")
```

Reply:
Mm no I think this is right how it is. You can't have 0 devices. 1 device means use a single GPU if available, otherwise use CPU. Otherwise, if we treated 0 devices as "use CPU", then the unique ID of the step would be different for CPU vs GPU, which we decided was not the right behavior.

Reply:
Yes, you are right.